polkadot/node/core/dispute-coordinator/src/tests.rs

```diff
@@ -2542,6 +2583,353 @@ fn issue_local_statement_does_cause_distribution_but_not_duplicate_participation
 	});
 }
 
+#[test]
+fn participation_with_onchain_disabling() {
```

Comment:
> Firstly confirmation should work even if ALL the votes are coming from disabled guys simply as a defence in depth against something going really badly within the system.

Does this mean that if we have got a dispute storm, as the last incident on Kusama, disabling won't save us?

Reply:
Re: confirmation trumping disablement I've opened a new issue because there is some extra analysis needed there. 

Nevertheless.... generally what happened in the recent kusama parachain stall would not be fully handled by the disabling strategy. We can deal with some dispute storms but not all dispute storms are alike. The recent one was caused by the faulty pruning logic which was present in all honest nodes. All honest nodes participated in something they shouldn't have. No disabling strategy can save us from that point so the best we could have done is freeze parachains and fix the critical error before the security is fully compromised.

If every node malfunctions no disabling will ever save us.

Reply:
>  All honest nodes participated in something they shouldn't have.

Valid point.

Reply:
Take a look at [49c066f](https://github.com/paritytech/polkadot-sdk/pull/2637/commits/49c066ff0ae1a365d225bcdb98dbdca89590bd75)